File: javascript/hamming/hamming.js
'use strict';

exports.compute = function(strandA, strandB) {
  var difference = 0;

  if (strandA.length !== strandB.length) {
    throw "DNA strands must be of equal length.";
  };

  for (var i=0; i < strandA.length; i++) {
    if (strandA.charAt(i) !== strandB.charAt(i)) {
      ++difference;
    };
  };

  return difference;
};

File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(strand_A, strand_B)
  difference = 0

  for i = 1, math.min(#strand_A, #strand_B) do
    if strand_A:sub(i, i) ~= strand_B:sub(i, i) then
      difference = difference + 1
    end
  end

  return difference
end

return Hamming
